### ISSUE TITLE
Update dependencies in package.json to latest and add peerDependencies

### ISSUE THREAD

**Issue by peterkc** [box/leche#4]
```javascript
{
  "peerDependencies": {
    "mocha": ">=1.18 <3"
  },
  "dependencies": {
    "mocha": "^2.1.0"
  },
  "devDependencies": {
    "browserify": "^8.0.2",
    "chai": "^1.9.1",
    "eslint": "^0.11.0",
    "istanbul": "^0.3.2",
    "jsdoc": "^3.3.0-alpha9",
    "jsonlint": "^1.6.2",
    "mockery": "~1.4",
    "shelljs": "^0.3.0",
    "sinon": "^1.10.3"
  }
}

**Comment by nzakas** [comment]
Can you explain what you are suggesting and why?

**Comment by nzakas** [comment]
I'm going to look at how switching to peerDependencies would affect existing projects.

**Comment by peterkc** [comment]
Thanks @nzakas. Sorry for the delayed replied. Going forward will split up my PR. Looking forward to the peerDependencies changes.

**Comment by nzakas** [comment]
No worries, the change has been merged in. I'm working on #2 at the moment, and when that's done, we'll do another release.

**Comment by nzakas** [comment]
Turns out #2 will be a breaking change, so I'm going to push out the peerDependencies change as a patch first.

**Comment by nzakas** [comment]
Published 1.0.3 with this change.
